### PR TITLE
feat: property-based scheduling: enable the property-based scheduling experience

### PR DESCRIPTION
**Pull request by michaelawyu**
### Description of your changes

This PR enables the property-based scheduling experience. It also adds additional E2E tests for the experience.

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

- [x] E2E tests

### Special notes for your reviewer

No including all the tests for PR size reasons; additional tests will be submitted in separate PRs.

### CODE REVIEW

File: pkg/propertyprovider/aks/provider.go [hunk 1]
```diff
@@ -119,25 +118,32 @@ func (p *PropertyProvider) Start(ctx context.Context, config *rest.Config) error
...
-		//
-		// This incurs the slightly higher overhead, however, as auto-discovery runs only
-		// once, the performance impact is negligible.
-		discoveredRegion, err := p.autoDiscoverRegionAndSetupTrackers(ctx, mgr.GetAPIReader())
-		if err != nil {
-			klog.ErrorS(err, "Failed to auto-discover region for the AKS property provider")
-			return err
+	klog.V(2).Info("Setting up the node tracker")
+	if p.nodeTracker == nil {
+		klog.V(2).Info("No pricing provider is set up; using the default AKS Karpenter pricing client")
```
Comment (ryanzhang-oss) [inline pkg/propertyprovider/aks/provider.go:123]: why is this related to nodeTracker?

File: test/e2e/setup.sh [hunk 2]
```diff
@@ -15,7 +15,7 @@ declare -a MEMBER_CLUSTERS=()
 
 for (( i=1;i<=MEMBER_CLUSTER_COUNT;i++ ))
 do
-  MEMBER_CLUSTERS+=("cluster-$i")
+    MEMBER_CLUSTERS+=("cluster-$i")
```
[the remark this answers is not in the document]
Comment (michaelawyu) [reply test/e2e/setup.sh:18]: Oh, sorry. It must be auto-linting; I will revert the change.

File: pkg/propertyprovider/aks/provider.go (hunk 1 above)
Comment (michaelawyu) [reply pkg/propertyprovider/aks/provider.go:123]: The pricing provider is actually a part of the node tracker; I will update the log entry to be clearer.

File: charts/member-agent/README.md [hunk 3]
```diff
@@ -40,5 +40,7 @@ helm upgrade member-agent member-agent/ --namespace fleet-system
 | resources                | The resource request/limits for the container image   | limits: "2" CPU, 4Gi, requests: 100m CPU, 128Mi |
 | namespace                | Namespace that this Helm chart is installed on.       | `fleet-system`                                  |
 | logVerbosity             | Log level. Uses V logs (klog)                         | `3`                                             |
+| propertyProvider         | The property provider to use with the member agent    | ``                                              |
```
Comment (ryanzhang-oss) [inline charts/member-agent/README.md:43]: does this mean the default is no provider?

File: pkg/propertyprovider/aks/provider.go [hunk 4]
```diff
@@ -119,25 +118,32 @@ func (p *PropertyProvider) Start(ctx context.Context, config *rest.Config) error
...
-		// has not been initialized yet and as a result cached client is not yet available.
-		//
-		// This incurs the slightly higher overhead, however, as auto-discovery runs only
-		// once, the performance impact is negligible.
-		discoveredRegion, err := p.autoDiscoverRegionAndSetupTrackers(ctx, mgr.GetAPIReader())
-		if err != nil {
-			klog.ErrorS(err, "Failed to auto-discover region for the AKS property provider")
-			return err
+	klog.V(2).Info("Setting up the node tracker")
+	if p.nodeTracker == nil {
```
Comment (ryanzhang-oss) [inline pkg/propertyprovider/aks/provider.go:122]: when will the node Tracker be not nil?

**Review by ryanzhang-oss — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -116,3 +116,10 @@ jobs:
         env:
           KUBECONFIG: '/home/runner/.kube/config'
           HUB_SERVER_URL: 'https://172.19.0.2:6443'
+          # Temporarily enable the AKS property provider for the E2E tests, in order
+          # to verify the property-based scheduling experience.
+          #
+          # TO-DO (chenyu1): to ensure a vendor-neutral experience, switch to a dummy
+          # property provider once the AKS one is split out.
+          PROPERTY_PROVIDER: 'aks'
```
[the remark this answers is not in the document]
Comment (michaelawyu) [reply .github/workflows/ci.yml:124]: Done.

File: charts/member-agent/README.md (hunk 3 above)
Comment (michaelawyu) [reply charts/member-agent/README.md:43]: Yeah, the default is the fallback behavior (only resource usage is collected, directly by the member agent). I'll update the description to be clearer.

File: pkg/propertyprovider/aks/provider.go (hunk 4 above)
Comment (michaelawyu) [reply pkg/propertyprovider/aks/provider.go:122]: Ah, this is for the case where a pricing provider is plugged in directly. Node tracker will be created here in the New function if no pricing provider is provided.

**Review by ryanzhang-oss — APPROVED** [review]
